GameTest Framework

Mireole opened the pull request:
# Pages sur le GameTest Framework
3 pages sur le GameTest Framework, une sur le framework en général, une sur les Batch(pour grouper les tests) et une autre sur les GameTestGenerator (nom explicite je pense)
## Page 1 : Utilisation du framework en général
![gametest_page_1](https://user-images.githubusercontent.com/83138584/160283236-19004f64-c00b-4427-bcc8-e57a238d82e2.png)

## Page 2 : Utilisation des Batch pour grouper des tests
![gametest_page_2](https://user-images.githubusercontent.com/83138584/160283243-133a37d8-57c4-4d99-9b8d-c0db4a8f28bf.png)

##Page 3 : Les GameTestGenerator
![gametest_page_3](https://user-images.githubusercontent.com/83138584/160283249-cee4c423-6cca-4c29-97be-7c516b3e9a11.png)

Mireole committed:
GameTest Framework, 1ere version

Mysterious-Dev reviewed (commented):
Peut être déplacer le dossier gametests dans advanced (pour ne pas recréer une catégorie de plus mais une "sous catégorie")

Mireole committed:
Passage de la catégorie dans la catégorie avancée en sous-dossier

Mysterious-Dev commented on docs/gametests/gametest-framework.md:
```diff
@@ -0,0 +1,182 @@
+
+Une fois le ou les tests lancés, vous devriez voir les templates être ajoutées au monde et les tests s'exécuter.
+
+Si un test succède, un beacon vert apparaîtra proche de la template.
```
_Si un test est un succès_ ?

Mireole committed:
Si un test succède -> si un test est un succès

Mysterious-Dev reviewed (approved):
Peut-être renommer le fichier `les_batch.md` en `batches.md`

Mireole committed:
les_batch.md -> batches.md

Mysterious-Dev commented on docs/advanced/gametests/gametest-framework.md:
```diff
@@ -0,0 +1,182 @@
+sidebar_position: 1
+title: Game Test Framework
+description: Comment utiliser le framework de test intégré à Forge ?
+tags: [GameTest]
```
Plutôt `game_test` que `GameTest` pour le tag

Mysterious-Dev commented on docs/advanced/gametests/batches.md:
```diff
@@ -0,0 +1,38 @@
+sidebar_position: 2
+title: Les Batch
+description: Comment utiliser les batchs pour grouper des GameTests ?
+tags: [GameTest]
```
Plutôt `game_test` que `GameTest` pour le tag

Mysterious-Dev commented on docs/advanced/gametests/gametest_generator.md:
```diff
@@ -0,0 +1,98 @@
+sidebar_position: 3
+title: GameTest Generator
+description: Comment utiliser les GameTest Generator ?
+tags: [GameTest]
```
Plutôt `game_test` que `GameTest` pour le tag

Mireole committed:
change tags

Mysterious-Dev commented on docs/advanced/gametests/gametest-framework.md:
```diff
@@ -0,0 +1,182 @@
+
+`/test create nomDuTest longueurDeLaStructure hauteurDeLaStructure profondeurDeLaStructure`
+
+(La taille, la largeur et la profondeur de la structure sont des valeurs optionnelles)
```
La taille ?? Plutôt la longueur au vu de la commande précedente

Mysterious-Dev commented on docs/advanced/gametests/gametest-framework.md:
```diff
@@ -0,0 +1,182 @@
+
+
+## Lancer les tests
+Pour lancer les tests et vérifier leur bon fonctionnement, vous avez deux solutions :
```
Plutôt _Pour lancer les tests et vérifier leurs bons fonctionnements, vous avez deux solutions :_ ?

Mireole committed:
La taille → La longueur
Pour lancer les tests et vérifier leur bon fonctionnement → Pour lancer les tests et vérifier leurs bons fonctionnements

Mysterious-Dev reviewed (commented):
Peut-être renommer gametest-framework.md ->framework.md et gametest_generator.md -> generator.md pour plus d'unicité avec les URL des autres catégories (éviter de répéter les mots dans l'URL)

Mireole committed:
position : 15 -> 4
gametest-framework.md -> framework.md
gamestest_generator.md -> generator.md

Mysterious-Dev commented on docs/advanced/gametests/framework.md:
```diff
@@ -0,0 +1,182 @@
+:::
+
+:::caution
+L'implémentation de ce système pour les moddeurs ayant été ajoutée dans dans Forge 39.0.88, vous devez être dans une version supérieure ou égale à celle-ci pour l'utiliser. (Je vous invite cependant à utiliser la dernière version, car il y a eu des changements dans ce système depuis son implémentation)
```
Mettre le . après la parenthèse (plus logique)

Mysterious-Dev commented on docs/advanced/gametests/framework.md:
```diff
@@ -0,0 +1,182 @@
+:::caution
+L'implémentation de ce système pour les moddeurs ayant été ajoutée dans dans Forge 39.0.88, vous devez être dans une version supérieure ou égale à celle-ci pour l'utiliser. (Je vous invite cependant à utiliser la dernière version, car il y a eu des changements dans ce système depuis son implémentation)
+
+Vous devez aussi vérifier que pour chaque configuration de run dans votre build.gradle, vous avez bien la ligne `property 'forge.enabledGameTestNamespaces', 'votre_mod_id'`. Cette ligne active tous les tests liés à votre modid.
```
_Chaque configuration de lancement_ non (question à débattre) ?

_vous avez bien la ligne `property 'forge.enabledGameTestNamespaces', 'votre_mod_id'`. Cette ligne active tous les tests liés à votre modid._ -> _vous avez bien la ligne suivante `property 'forge.enabledGameTestNamespaces', 'votre_mod_id'`, celle-ci activant tous les tests liés à votre modid._ ?

Mireole replied:
pour configuration de lancement effectivement j'aurais dû traduire effectivement je pense, je change ça tout de suite

Mysterious-Dev commented on docs/advanced/gametests/framework.md:
```diff
@@ -0,0 +1,182 @@
+
+Vous devez aussi vérifier que pour chaque configuration de run dans votre build.gradle, vous avez bien la ligne `property 'forge.enabledGameTestNamespaces', 'votre_mod_id'`. Cette ligne active tous les tests liés à votre modid.
+
+Enfin, si vous ne l'avez pas, je vous invite à ajouter la configuration de run `runGameTestServer` avec les autres, qui va lancer un serveur sur votre ordinateur qui va exécuter tous les tests puis se fermer. Vous pouvez trouver plus d'informations dessus dans la section **Exécuter les tests**.
```
_qui va lancer un serveur sur votre ordinateur qui va exécuter tous les tests puis se fermer._ -> _celle-ci va lancer un serveur sur votre ordinateur qui va exécuter tous les tests puis se fermer._ ?

Mysterious-Dev commented on docs/advanced/gametests/framework.md:
```diff
@@ -0,0 +1,182 @@
+`/test create nomDuTest longueurDeLaStructure hauteurDeLaStructure profondeurDeLaStructure`
+
+(La longueur, la largeur et la profondeur de la structure sont des valeurs optionnelles)
+Le nom du test n'est pas important, car il devra être changé plus tard de toute façon.
```
Pas forcement d'accord. Celui-ci est normalement important car il est aussi stocké dans le structure block du template (après je ne sais pas si tu as fait des tests pour vérifier cela (je peux me tromper aussi)). C'est à débattre bien entendu.

Mireole replied:
Lorsque la template est posée (pour les tests pas exemple, il reprend le nom du fichier donc change celui stocké dans le structure block, qui n'est d'ailleurs pas inclus dans la template si c'est bien la question (j'ai peut-être mal compris)

Mysterious-Dev replied:
C'est un truc à tester, je regarderais de mon coté et je te redirais.

ZomDev3343 reviewed (approved):
Les pages sont bien écrites, pour moi c'est bon.

Mireole committed:
Changements du texte
configuration de run -> configuration de lancement
et les 2 autres suggérés pas Mysterious_Dev

Mireole committed:
déplacement d'un point

Mireole committed:
Changement des noms des pages
